asyncstatsd/metrics/datadog.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass, field
+from datetime import timedelta
+from typing import Dict
+
+from .basic import GaugeMetricMixin, StatsdMetric, TimingMetricMixin
+
+
+@dataclass
+class DatadogMetric(StatsdMetric):
+    tags: Dict | None = field(default_factory=dict)
+
+    def _build_tags(self) -> str:
+        return ",".join(f"{k}:{v}" for k, v in self.tags.items())  # type: ignore
```

Comment:
Same question about validating tag keys/values to ensure they don't include special chars that break serialisation, like `:`